packages/components/src/duotone-picker/duotone-picker.tsx

```diff
@@ -130,10 +130,27 @@ function DuotonePicker( {
 	let metaProps:
 		| { asButtons: false; loop?: boolean; 'aria-label': string }
 		| { asButtons: false; loop?: boolean; 'aria-labelledby': string }
-		| { asButtons: true };
+		| { asButtons: true; 'aria-label': string }
+		| { asButtons: true; 'aria-labelledby': string };
 
 	if ( asButtons ) {
-		metaProps = { asButtons: true };
+		const _metaProps: { asButtons: true } = {
+			asButtons: true,
+		};
+
+		if ( ariaLabel ) {
+			metaProps = { ..._metaProps, 'aria-label': ariaLabel };
+		} else if ( ariaLabelledby ) {
+			metaProps = {
+				..._metaProps,
+				'aria-labelledby': ariaLabelledby,
+			};
+		} else {
+			metaProps = {
+				..._metaProps,
+				'aria-label': __( 'Custom color picker.' ),
+			};
+		}
```

Comment:
The whole logic for building the `metaProps` object feels more complicated than it could be, potentially with less `if` statements and without using an intermediate `_metaProps` object.

I don't have much to take a look at it today, but if needed I can try to help in the next days.